api_server.py

```
@@ -27,9 +27,22 @@
+            g.db = sqlite3.connect(DB_PATH, detect_types=sqlite3.PARSE_DECLTYPES)
+            g.db.row_factory = sqlite3.Row
+        return g.db
+    except RuntimeError:
+        # Fallback for when called outside of a Flask application context
+        conn = sqlite3.connect(DB_PATH, detect_types=sqlite3.PARSE_DECLTYPES)
+        conn.row_factory = sqlite3.Row
+        return conn
+
+
+@app.teardown_appcontext
+def close_db(error):
```

Ruff (0.15.1)

[warning] Unused function argument: `error`

(ARG001)